packages/boxel-ui/addon/src/components/drag-and-drop/index.gts

```diff
@@ -0,0 +1,259 @@
+import { fn, hash } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import type { ComponentLike } from '@glint/template';
+
+//Dnd Kanban Board
+export interface ColumnHeaderArgs {
+  title?: string;
+}
+
+export interface DndKanbanBoardArgs<Column> {
+  columnHeader?: ComponentLike<ColumnHeaderArgs>;
+  columns: Column[];
+}
+
+export interface DndKanbanBoardSignature<Column> {
+  Args: DndKanbanBoardArgs<Column>;
+  Blocks: {
+    default: [card: Card, column: Column];
+  };
+}
+
+export class Card {
+  [key: string]: any;
+
+  constructor(...args: any[]) {
+    if (args.length === 1 && typeof args[0] === 'object') {
+      Object.assign(this, args[0]);
+    }
+  }
+}
+
+export class Column {
+  @tracked title: string;
+  @tracked cards: Card[];
+
+  constructor(title: string, cards: Card[] = []) {
+    this.title = title;
+    this.cards = cards;
+  }
+}
+
+export default class DndKanbanBoard extends Component<
+  DndKanbanBoardSignature<Column>
+> {
+  @tracked columns: Column[] = this.args.columns;
+  @tracked hoveredColumnIndex: number | null = null;
+
+  get isDropZoneTargeted() {
+    return (columnIndex: number) => this.hoveredColumnIndex === columnIndex;
+  }
+
+  @action onColumnDragEnter(columnIndex: number) {
+    this.hoveredColumnIndex = columnIndex;
+  }
+
+  @action onColumnDragLeave() {
+    this.hoveredColumnIndex = null;
+  }
+
+  @action clearHoveredState() {
+    this.hoveredColumnIndex = null;
+  }
+
+  @action moveColumn({
+    source: { data: draggedItem },
+    target: { data: dropTarget, edge },
+  }: any) {
+    this.columns = removeItem(this.columns, draggedItem);
+
+    if (edge === 'left') {
+      this.columns = insertBefore(this.columns, dropTarget, draggedItem);
+    } else {
+      this.columns = insertAfter(this.columns, dropTarget, draggedItem);
+    }
+  }
+
+  @action moveCard({
+    source: {
+      data: { item: draggedItem, parent: draggedItemParent },
+    },
+    target: {
+      data: { item: dropTarget, parent: dropTargetParent },
+      edge,
+    },
+  }: any) {
+    draggedItemParent.cards = removeItem(draggedItemParent.cards, draggedItem);
+
+    if (!dropTarget) {
+      dropTargetParent.cards = insertAt(dropTargetParent.cards, 0, draggedItem);
+    } else if (edge === 'top') {
+      dropTargetParent.cards = insertBefore(
+        dropTargetParent.cards,
+        dropTarget,
+        draggedItem,
+      );
+    } else {
+      dropTargetParent.cards = insertAfter(
+        dropTargetParent.cards,
+        dropTarget,
+        draggedItem,
+      );
+    }
+
+    this.clearHoveredState();
+  }
+
+  <template>
+    <div class='draggable-container' {{on 'dragend' this.clearHoveredState}}>
+      {{#each this.columns as |column columnIndex|}}
+        <div
+          class='column'
+          {{DndDropTargetModifier
```

Comment:
if we are in fastboot we will not want to include this modifier

Reply:
Dynamic imports seems like a weird way to solve this problem. I think we could use [the `modifier` helper](https://api.emberjs.com/ember/5.12/classes/Ember.Templates.helpers/methods/modifier?anchor=modifier) to conditionally apply a modifier when not in Fastboot.

Reply:
yeah, i don't think we have a choice, the underlying lib that ember-drag-and-drop uses performs has side effects that need the browser context when the modules are evaluated. we have to dynamically load these modules the very very act of importing them runs the side effects that blow up fastboot